src/main/webapp/components/UploadForm.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <v-form ref="form" @submit.prevent="onSubmit">
+    <v-row>
+      <v-col>
+        <v-file-input label="Upload a PDF"
+                      v-model="file"
+                      accept=".pdf"
+                      show-size
+                      @change="onFileChange">
+        </v-file-input>
+
+        <v-text-field label="Title"
+                      v-model="title"
+                      clearable>
+        </v-text-field>
+
+        <v-combobox label="School"
+                    v-model="school"
+                    ref="school"
+                    clearable chips
+                    :items="commonSchools">
+          <template v-slot:no-data>
+            <span v-html="noDataHtml"></span>
+          </template>
+        </v-combobox>
+
+        <v-combobox label="Course"
+                    v-model="course"
+                    ref="course"
+                    clearable chips
+                    :items="commonCourses">
+          <template v-slot:no-data>
+            <span v-html="noDataHtml"></span>
+          </template>
+        </v-combobox>
+      </v-col>
+
+      <v-col>
+        <iframe :src="previewUrl"></iframe>
+      </v-col>
+    </v-row>
+
+    <v-combobox label="Miscellaneous Labels"
+                v-model="miscLabels"
+                multiple chips deletable-chips
+                clearable hide-selected
+                :items="commonLabels">
+      <template v-slot:no-data>
+        <span v-html="noDataHtml"></span>
+      </template>
+    </v-combobox>
+
+    <v-btn type="submit">Post Note</v-btn>
+  </v-form>
+</template>
+
+<script>
+module.exports = {
+  data: function() {
+    return {
+      uploadUrl: null,
+      file: null,
+      previewUrl: null,
+      title: null,
+      school: "Unaffiliated",
+      commonSchools: ['UF', 'MIT', 'NYU', 'UCLA'],
+      course: "Unaffiliated",
+      commonCourses: ['CIS4301', 'ENC1101', 'PHY2049'],
+      miscLabels: null,
+      commonLabels: ['Hard', 'Professor X', 'Test Prep'],
+      noDataHtml: "No matching results. Type and press <kbd>enter</kbd> to create a new one",
+    }
+  },
+  watch: {
+    // Blur input fields for school and course when a selection is made.
+    // Communicates to the user that their custom input is selected.
+    // Must wait for next tick to avoid race conditions.
+    school: function() {
+      this.nextTick(() => this.$refs.school.blur());
+    },
+    course: function() {
+      this.nextTick(() => this.$refs.course.blur());
+    },
+  },
+  methods: {
+    nextTick: function(f) {
+      // Avoids race conditions with function f
+      setTimeout(f, 0);
```

Comment:
Could you possibly add an image of how the UI is affected by the race condition this is preventing?

Reply:
As a brief description:
![image](https://user-images.githubusercontent.com/44207690/87602652-6430d400-c6c5-11ea-8071-6dd38fe8c16b.png)
Without the timeout, clicking on the MIT suggestion or pressing Enter/Tab at this point wouldn't work. The current search, 'M', would become the label because the input would blur before getting the new MIT suggestion, and the component sets the current search input as the label on blur. So by delaying the blur just a little, the MIT suggestion has time to become the new label.

Reply:
Woahhh this is very interesting, I wonder how other sites have implemented it and if they also have to account for this race condition with the blur. I think it's a great solution for the time being since it works and looks epic but is definitely something we could loop back to if we develop a better understanding of how to implement this functionality.

Reply:
Unfortunately I'm not familiar enough with vue or even native js to know but yes, might be a worthwhile thing to look up, as I imagine you're not the first person doing this!

Reply:
We would do something similar a lot in angularJS using the $timeout service to delay execution of various threads. Thanks for documenting why you are doing this and this could be the recommended way of doing things but's definitely worth double checking.